managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImpl.java

```diff
@@ -301,11 +302,17 @@ public void asyncRemoveCursor(String ledgerName, String cursorName, MetaStoreCal
                     }
                     callback.operationComplete(null, null);
                 }, executor.chooseThread(ledgerName))
-                .exceptionally(ex -> {
+                .exceptionallyAsync(ex -> {
+                    Throwable actEx = FutureUtil.unwrapCompletionException(ex);
+                    if (actEx instanceof MetadataStoreException.NotFoundException){
+                        log.info("[{}] [{}] cursor delete done because it did not exist.", ledgerName, cursorName);
+                        callback.operationComplete(null, null);
+                        return null;
+                    }
                     executor.executeOrdered(ledgerName, SafeRunnable.safeRun(() -> callback
                             .operationFailed(getException(ex))));
                     return null;
-                });
+                }, executor.chooseThread(ledgerName));
```

Comment:
We can remove `executor.executeOrdered` from line 312?

Reply:
Good suggestion, fixed